src/conn/conn_api.c

```diff
@@ -1022,6 +1022,112 @@ err:	API_END_RET_NOTFOUND_MAP(session, ret);
 }
 
 /*
+ * __conn_optrack_dir --
+ *	Set the directory for operation logging
+ */
+static int
+__conn_optrack_dir(WT_SESSION_IMPL *session, const char *cfg[])
+{
+	WT_CONFIG_ITEM cval;
+
+	/* Only use the environment variable if configured. */
+	WT_RET(__wt_config_gets(session, cfg, "use_environment", &cval));
+	if (cval.val != 0 &&
+	    __wt_getenv(session, "WIREDTIGER_OPTRACK",
+			&S2C(session)->optrack) == 0)
+		return (0);
+
+	/*
+	 * If there's no WIREDTIGER_OPTRACK environment variable, use the
+	 * configured database directory.
+	 */
+	S2C(session)->optrack = ".";
+
+	return (0);
+}
+
+/*
+ * A global variable that tells us whether or not we have created a
+ * map file for the current process. Map files, translating binary
+ * numbers into function names are valid per-process, so we have to
+ * set them once per process, not per connection or per session.
+ */
+static bool __wt_optrack_map_setup = 0;
+
+/*
+ * __conn_optrack_setup --
+ *     Set up operation logging.
+ */
+static int
+__conn_optrack_setup(WT_SESSION_IMPL *session, const char *cfg[])
+{
+	WT_CONNECTION_IMPL *conn;
+	char optrack_map_name[PATH_MAX];
+	bool exists;
+
+	conn = S2C(session);
+
+	/* Set up the directory for operation logs. */
+	WT_RET(__conn_optrack_dir(session, cfg));
+
+	/* Operation tracking files will include the ID
+	 * of the creating process in their name, so we
+	 * can distinguish between log files created by
+	 * different WiredTiger processes in the same
+	 * directory. We cache the process id for future use.
+	 */
+	conn->optrack_pid = __wt_process_id();
```

Comment:
With operation tracking path being always relative to a wiredtiger directory would this be needed?
We wont be having two WT instances dumping to same directory, but a restart will change the id.

Reply:
I believe it is useful - for example if we used this feature with wtperf, this means we get a different set of files for the load and workload phase(s).